src/cryptography/x509.py

```diff
@@ -104,6 +104,27 @@ def __repr__(self):
     dotted_string = utils.read_only_property("_dotted_string")
 
 
+class Name(object):
+    def __init__(self, attributes):
+        self._attributes = attributes
+
+    def get_attributes_for_oid(self, oid):
+        return [i for i in self._attributes if i.oid == oid]
+
+    @property
+    def attributes(self):
+        return self._attributes[:]
```

Comment:
What if instead of a `.attributes`, we made `Name` objects be iterable; `for x in subject.issuer:`, does that make sense, or no?